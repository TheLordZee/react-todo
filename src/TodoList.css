.TodoList {
    border: 4px solid black;
    border-radius: 15%;
    padding: 5vh;
    height: 100vh;
}