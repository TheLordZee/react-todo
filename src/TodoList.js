import React, {useState} from "react";
import {v4 as uuid} from 'uuid';
import Todo from "./Todo";
import TodoForm from "./TodoForm";
import "./TodoList.css"

const TodoList = () => {
    const INITIAL_STATE = {
        todoText: ""
    }
    const [list, setList] = useState([]);
    const [formData, setFormData] = useState(INITIAL_STATE);

    const addTodo = (todoData) => {
        const todo = {...todoData, id:uuid()}
        setList(td => [...td, todo])
    }

    const removeTodo = (id) => {
        let newList = [...list]
        for(let i = 0; i < newList.length; i++){
            if(newList[i].id === id){
                newList.splice(i,1)
            }
        }
        setList(newList)
    }

    const onChange = e => {
        let {value} = e.target;
        setFormData({todoText: value})
    }

    const handleSubmit = e => {
        e.preventDefault();
        addTodo(formData);
        
        setFormData(INITIAL_STATE);
    }

    return(
        <div className="TodoList">
            <TodoForm 
                handleSubmit={handleSubmit} 
                onChange={onChange} 
                formData={formData}
            />
            <ul className="TodoList-todos">
                {list.map(td =>
                    <Todo key={td.id} text={td.todoText} removeTodo={() => removeTodo(td.id)}/>
                )}
            </ul>
        </div>
    )
}

export default TodoList;