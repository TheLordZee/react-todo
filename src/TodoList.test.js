import { fireEvent, render } from '@testing-library/react';
import TodoList from "./TodoList"

it('renders', () => {
  render(<TodoList />);
});

it("matches snapshot", () => {
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
})

it("can add a new todo", () => {
    const {queryByText, container} = render(<TodoList />);
    const textInput = container.getElementsByClassName('TodoForm-input')[0];
    const submitBtn = queryByText("Add Todo")

    fireEvent.change(textInput, {target: {value: "Test todo"}})  
    fireEvent.click(submitBtn)

    expect(queryByText("Test todo")).toBeInTheDocument()
})

it("can delete a todo", () => {
    const {queryByText, container} = render(<TodoList />);
    const textInput = container.getElementsByClassName('TodoForm-input')[0];
    const submitBtn = queryByText("Add Todo")

    fireEvent.change(textInput, {target: {value: "Test todo"}})  
    fireEvent.click(submitBtn)
    let todo = queryByText("Test todo")

    expect(todo).toBeInTheDocument()

    const deleteBtn = queryByText("X")
    fireEvent.click(deleteBtn);
    expect(todo).not.toBeInTheDocument();
})