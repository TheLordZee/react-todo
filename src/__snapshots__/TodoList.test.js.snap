// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <form
      class="TodoForm"
    >
      <label
        for="text"
        hidden=""
      >
        Text
      </label>
      <input
        class="TodoForm-input"
        name="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <ul
      class="TodoList-todos"
    />
  </div>
</DocumentFragment>
`;
